one-ui/src/main/java/com/marklogic/hub/oneui/services/DataHubProjectUtils.java

```diff
@@ -140,4 +167,62 @@ public static void extractZipProject(HubProject project, InputStream in, UIDeplo
         }
         listener.onStatusChange(0, String.format("Completed extracting the uploaded zip project into (%s)", currProjectDir.toFile().getAbsolutePath()));
     }
+
+    private static InputStream toByteArrayInputStream(InputStream in) {
+        if (in instanceof ByteArrayInputStream) {
+            return in;
+        }
+        byte[] buff = new byte[8000];
+        int bytesRead = 0;
+        ByteArrayOutputStream bao = new ByteArrayOutputStream();
+        try {
+            while ((bytesRead = in.read(buff)) != -1) {
+                bao.write(buff, 0, bytesRead);
+            }
+        } catch (IOException e) {
+            throw new RuntimeException(String.format("Failed to convert to ByteArrayInputStream due to (%s)", e.getMessage()));
+        }
+        byte[] data = bao.toByteArray();
+        ByteArrayInputStream bin = new ByteArrayInputStream(data);
+        return bin;
+    }
+
+    private static String getArchiveFolderOfZipFile(Path currProjectDir, InputStream in) {
```

Comment:
I highly recommend unit tests to verify this functionality. There's a lot of code being added here, and the new test in EnvironmentControllerTest verifies it at an integration level. But if something goes wrong, it'll be difficult to debug because we don't have any unit tests verifying any of the new logic. 

Looking at what this method outputs, it's not clear to me what this method is doing. The name implies an "archive folder" is being returned, so I'm first wondering why a Path or File isn't returned. I also see "return null" at the bottom - what does that mean if that happens? Unit tests would both verify that this functionality works correctly and document the purposes of the method as well.